Singular/svd_si.h

```diff
@@ -3224,7 +3224,7 @@ namespace bidiagonal
         {
             return;
         }
-        ap::ap_error::make_assertion(fromtheright && zcolumns==m || !fromtheright && zrows==m);
+        ap::ap_error::make_assertion(fromtheright ? zcolumns==m : zrows==m);
```

Comment:
Note that `||` bind stronger than `&&` binds stronger than `==`, so the old code checked `(fromtheright && zcolumns)== ((m || !fromtheright) && zrows) ==m`